batchlib/base.py

```diff
@@ -303,3 +305,29 @@ def __init__(self, input_pattern, input_key, output_key,
                          input_key=input_key, output_key=output_key,
                          input_ndim=input_ndim, output_ndim=output_ndim,
                          target=target)
+
+
+class BatchJobWithSubfolder(BatchJob):
+
+    """ Base class for batch jobs that output into a folder
+    """
+
+    def __init__(self, *args, output_folder="", **kwargs):
+        self.output_folder = output_folder
+
+        super().__init__(*args, **kwargs)
```

Comment:
It's convenient, but I would not use the *args, **kwargs pattern here, because not all kwargs of the super class make sense here: `output_keys` and `output_dim` don't make sense here. This means arguments should also be passed to this classes `__init__` directly.
(As discussed, I think these should probably not be part of the super class, but rather go only to `BatchJobOnContainer`, but I need some thinking time to clean up the design properly.)